extensions/items/standalone.mjs

```diff
@@ -261,28 +281,72 @@ class Standalone {
    */
   async c2s_dropItem(playerID, mapCoord, facing, itemName) {
     /* player has not registered yet */
-    if (!this.items.has(playerID)) {
-      console.error('Player not initialized');
-      return;
+    if (!(playerID in this.items)) {
+      this.items[playerID] = {};
     }
 
-    /* update database */
-    let itemObj = this.items.get(playerID);
+    let itemObj = this.items[playerID];
     if (!itemName in itemObj) {
       console.error('No such item');
       return;
     }
+    if (!this.itemInfo[itemName].droppable) {
+      console.log("Item is not droppable");
+      return;
+    }
     itemObj[itemName].amount -= 1;
-    this.items.set(playerID, itemObj);
+    this.items[playerID, itemObj];
 
     /* update cell set */
-    /* TODO: calculate dropped coordinate. Currently, the dropped coordinate is the same as the player's map coordinate */
-    this.droppedItemCell.set(this.droppedItemIndex, { "x": mapCoord.x, "y": mapCoord.y, "w": 1, "h": 1 });
-    this.droppedItemInfo.set(this.droppedItemIndex, itemName);
-    this.droppedItemIndex++;
-    for (const mapName of this.helper.gameMap.getOriginalCellSetStartWith('').keys()) { // Get all available maps.
-      this.gameMap.updateDynamicCellSet(mapName, 'droppedItem' + itemName.toUpperCase(), Array.from(this.droppedItemCell.values()));
+    const mapSize = this.gameMap.getMapSize(mapCoord.mapName);
+    let droopedItemCoord = {}
+    switch (facing) {
+      case 'U':
+        if (mapCoord.y === 0) {
+          droppedItemCoord.y = mapCoord.y + 1; // dropped back instead of front
+        }
+        else {
+          droppedItemCoord.y = mapCoord.y - 1;
+        }
+        droppedItemCoord.x = mapCoord.x;
+        break;
+      case 'D':
+        if (mapCoord.y === mapSize.y - 1) {
+          droppedItemCoord.y = mapCoord.y - 1; // dropped back instead of front
+        }
+        else {
+          droppedItemCoord.y = mapCoord.y + 1;
+        }
+        droppedItemCoord.x = mapCoord.x;
+        break;
+      case 'L':
+        if (mapCoord.x === 0) {
+          droppedItemCoord.x = mapCoord.x + 1; // dropped right instead of left
+        }
+        else {
+          droppedItemCoord.x = mapCoord.x - 1;
+        }
+        droppedItemCoord.y = mapCoord.y;
+        break;
+      case 'R':
+        if (mapCoord.x === mapSize - 1) {
+          droppedItemCoord.x = mapCoord.x - 1; // dropped left instead of right
+        }
+        else {
+          droppedItemCoord.x = mapCoord.x + 1;
+        }
+        droppedItemCoord.y = mapCoord.y;
+        break;
     }
+    this.droppedItemCell[this.droppedItemIndex] = { "x": droppedItemCoord.x, "y": droppedItemCoord.y, "w": 1, "h": 1 };
+    this.droppedItemInfo[this.droppedItemIndex] = itemName;
+    this.droppedItemIndex++;
+
+    /* Store data into file */
+    this.helper.storeData(this._pack_stored_data());
```

Comment:
`this._pack_stored_data()` is async, do you want to `await` or just make it not async?

Reply:
The function should't be an async function. Sorry for the mistake. It's fixed now.